lib/maths/CDecompositionComponent.cc

```diff
@@ -136,6 +137,21 @@ TDoubleDoublePr CDecompositionComponent::value(double offset, double n, double c
             return {m, m};
         }
 
+        // Sanity check that neither m nor sd are NaN values
+        // This is an extremely rare edge case as checks elsewhere prevent NaNs from
+        // being added to the models. As yet it is unknown how NaNs could be returned
+        // from the splines, only that they do occur.
+        // The returned interval will not contribute to the baseline.
+        if (maths::CMathsFuncs::isNan(m)) {
+            LOG_WARN(<< "Calculated mean value is not a number.");
+            return {0.0, 0.0};
```

Comment:
Maybe `{m_MeanValue, m_MeanValue}` would be a better fallback here.  That's what's returned when the spline is not initialized.

That implies it would be necessary to defend `m_MeanValue`, but that's probably a good thing anyway.  So maybe search for `m_MeanValue = ` in this file and log and error and refuse to set if the right hand side is `nan`.